doc/memory_usage.md

```diff
@@ -0,0 +1,133 @@
+Memory usage estimation for a module
+====================================
+
+This document aims to provide information useful to make a rough estimation
+of necessary memory to execute a WASM module.
+
+Instead of trying to cover every possible configurations,
+the following configuration is assumed in this document:
+
+* Module is built with `wasi-sdk`
+* Module is loaded with `wasm_runtime_load`
+* AOT is used
+* WASI is used
+* libc heap is used
+* app heap is not used
+* wasi-threads based pthread might be used
```

Comment:
worth mentioning `WASM_ENABLE_WASI_THREADS` just like you do in the line below for `lib_pthread`

Reply:
done